ibis/expr/tests/test_newrels.py

```diff
@@ -1567,3 +1568,31 @@ def test_subsequent_order_by_calls():
     first = ops.Sort(t, [t.int_col.desc()]).to_expr()
     second = ops.Sort(first, [first.int_col.asc()]).to_expr()
     assert ts.equals(second)
+
+
+def test_temporal_join():
+    t1 = ibis.table(name="t1", schema={"a": "int64", "b": "string", "c": "timestamp"})
+    t2 = ibis.table(name="t2", schema={"d": "int64", "e": "string"})
+
+    joined = t1.join(t2.at_time(t1.c), [t1.a == t2.d])
+
+    right = t2.at_time(t1.c)
+    assert right.op() == ops.VersionedTable(t2, at_time=t1.c)
+
+    r1 = ops.JoinTable(t1, index=0).to_expr()
+    r2 = ops.JoinTable(ops.VersionedTable(t2, at_time=r1.c), index=1).to_expr()
```

Comment:
An alternative implementation would be to have specialized join link (which might be considerable for asof joins as well), but we can defer that to a follow-up.